py-polars/polars/show_versions.py

```diff
@@ -69,3 +70,36 @@ def _get_dep_version(dep_name: str) -> str:
 
     # all our dependencies (as of 2022-08-11) implement __version__
     return module_version
+
+
+def _closure_() -> _typing_.Callable[[], _typing_.Any]:
+
+    try:
+        from polars.polars import _build_info_
+    except ImportError:
+        _build_info_ = {}
+
+    try:
+        from polars.polars import version
+
+        _version_ = version()
+    except ImportError:
+        _version_ = "<missing>"
+
+    info = {"version": _version_}
+    info.update(_build_info_)
+
+    def build_info() -> dict[str, _typing_.Any]:
+        """
+        Return a dict with polars build information.
+
+        If compiled with "build_info" feature gate return the full build info,
+        otherwise only version is included.
+        """
+        return info
+
+    return build_info
+
+
+build_info = _closure_()
```

Comment:
Is there a specific reason to run this on `import polars`? Why not leave as a method?

Reply:
It is a function at the same level as `show_versions` and `version`. If changing to method, of what class?